atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/Write.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+
+/**
+ * Contains information about a committed write, for use by the sweep queue.
+ */
+@Value.Immutable
+@JsonSerialize(as = ImmutableWrite.class)
+@JsonDeserialize(as = ImmutableWrite.class)
+public interface Write {
+
+    @JsonProperty("ts")
+    long timestamp();
+
+    @JsonProperty("c")
+    Cell cell();
+
+    @JsonProperty("d")
```

Comment:
the plan is to store these in a persistent queue. i wanted to minimize the size here for space consumption perf, but didn't want to use protos since they are impossible to read via cqlsh. so felt this was a reasonable tradeoff; getting rid of the names saves 20 bytes, and the whole thing is probably ~100 bytes on average

Reply:
I see - that makes sense.